### PR TITLE
create bot commands menu

### PR DESCRIPTION
**Pull request by WashingtonKayaker**

### CODE REVIEW

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md [hunk 1]
```diff
@@ -1,12 +1,167 @@
...
+# Bot command menus
 
+Creating a command menu enables you to give your users a list of commands that your bot is designed to execute, along with simple help text associated with each command that you can use to explain the command. The user can select the command from the list and it is placed in the message, then all they need to do is click __Send__.
+
+![Bot command menu](Media\bot-menu-sample.png)
+
+
+## Create a command menu for your bot
+
+Command menus are contained in your App Manifest, which can be created in <!--NEED LINK--> [App Studio](https://www.sadanduseless.com/funniest-animals-of-2019/), or edited directly in your source code in the `manifest.json` file in the `TeamsAppManifest` folder. In this section we will demonstrate how to create the command menu in App Studio, followed by the resulting `manifest.json` file so you can choose the approach that best suites your needs.
```
Comment (clearab) [inline msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: don't capitalize "app manifest"

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md [hunk 2]
```diff
@@ -1,12 +1,167 @@
...
+
+### Creating a command menu for your bot using App Studio
+
+The instructions here assume that you will be editing an existing app manifest, however the steps of adding a command menu are the same, whether you are creating a new manifest or editing an existing manifest.
```
Comment (clearab) [inline msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: Use "existing app manifest. The steps" or "existing app manifest. However, the steps"

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md [hunk 3]
```diff
@@ -1,12 +1,167 @@
...
+
+1. Open App Studio by selecting Apps from the lower left-hand side of your Teams window, and entering "App Studio" in the search box. If you do not have App Studio available you can download it, please refer to <!--NEED LINK--> [Installing App Studio](https://docs.microsoft.com/en-us/microsoftteams/platform/get-started/get-started-app-studio#installing-app-studio).
```
Comment (clearab) [inline msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: "Open App Studio from the ... overflow menu on the left navigation rail". No need to open the app store if it is already installed. Alternative you could use "/goto app studio" in the command box at the top of the screen.

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md [hunk 4]
```diff
@@ -1,12 +1,167 @@
...
+        }
+      ]
+    }
+  ],
+  ...
+}
+```
+
+
+## Handling Menu Commands in your Bot Code
```
Comment (clearab) [inline msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: Use sentence casing here, not title casing

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md [hunk 5]
```diff
@@ -1,12 +1,167 @@
...
+Bots in a group or channel respond only when they are mentioned ("@botname") directly in a message, and as a result every message received by a bot when in a group or channel scope will contains it own name in the message text returned (see `turnContext.Activity.Text`), and you need to ensure your message parsing handles that before handling the command being returned. You can parse out the _@Mention_ portion of the message text using a static method provided with the Microsoft Bot Framework, a method of the `Activity` class named `RemoveRecipientMention`.
+
+```csharp
+turnContext.Activity.RemoveRecipientMention();
+```
+
+
+
+
+## Command Menu Best Practices
```
Comment (clearab) [inline msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: Use sentence casing here, not title casing

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md [hunk 6]
```diff
@@ -1,12 +1,167 @@
...
+```
+
+#### Manifest excerpt - single menu for both scopes
+
+```json
+{
+  ...
+  "bots":[
+    {
+      "botId":"[Microsoft app ID for your bot]",
```
Comment (clearab) [inline msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: Use `<Microsoft app Id for your bot>` (don't use placeholder characters in JSON that are valid JSON characters, this makes it look like it is an array of Ids)

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md [hunk 7]
```diff
@@ -1,12 +1,167 @@
...
+
+Another valid approach for creating a commend menu is to create it directly in the manifest file while developing your bot source code. Here are a few things to keep in mind when using this approach:
+
+1. Each menu supports up to 10 commands.
+
+2. You can create a single command menu that will work in all scopes.
+
+3. You can create a different command menu for each scope
+
+#### Manifest excerpt - single menu for both scopes
```
Comment (clearab) [inline msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: Use example not excerpt

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md [hunk 8]
```diff
@@ -1,12 +1,167 @@
...
+          ]
+        }
+      ]
+    }
+  ],
+  ...
+}
+```
+
+#### Manifest excerpt - single menu for both scopes
```
Comment (clearab) [inline msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: use example not excerpt

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md [hunk 9]
```diff
@@ -1,12 +1,167 @@
...
+{
+  ...
+  "bots":[
+    {
+      "botId":"[Microsoft app ID for your bot]",
+      "scopes": [
+        "groupChat",
+        "team"
+      ],
+      "commandLists":[
```
Comment (clearab) [inline msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: In this case you'd want to define a different command in each scope. If you're defining the same command you wouldn't use two separate command lists.

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md (hunk 1 above)
Comment (clearab) [reply msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: This section doesn't clearly explain that you're talking about two options, and they only need to do one of them. Please refactor a bit to be more explicit.

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md (hunk 6 above)
Comment (WashingtonKayaker) [reply msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: I copied entire snippet from the existing page [Add a bot menu in Microsoft Teams](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/bots/bots-menus).

I made the suggested changes.

File: msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md (hunk 9 above)
Comment (WashingtonKayaker) [reply msteams-platform/bots/how-to/conversations/create-a-bot-commands-menu.md]: The "description" is different for each scope.

This section (Along with its title) come directly from [Add a bot menu in Microsoft Teams](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/bots/bots-menus#manifest-excerpt---separate-menu-per-scope).

**Pull request merged**